src/backend/distributed/sql/citus--9.3-2--9.2-2.sql

```diff
@@ -0,0 +1,32 @@
+-- citus--9.3-2--9.2-2
```

Comment:
This response is also related to @hanefi his line in the PR description:
> I plan to have all the migration files in the master and cherry-pick a sufficient subset to release branches so that future backport releases will contain valid downgrade paths.

I think it's better to have a separate `citus-downgrade` package that contains all the downgrade scripts and make the citus package depend upon it (I'm fine with a different package name). This has a few advantages:
1. when after upgrading from 9.2.0 to 9.3.0 you can actually downgrade to 9.2.0, instead of having to downgrade to 9.2.1, which contains the downgrade scripts from 9.3.0.
2. No need to do a lot of cherry picks to backport these changes to old releases. People can just install the shared `citus-backport` package for the version they need to backport from.

This works, because the SQL files don't change accross releases. They are only added (and maybe removed, I forgot what we did with the bad 9.2.3 release).

NOTE: It might be possible to put all the upgrade scripts in the same package as well, but I have the feeling this has would result in complications that are not worth the effort to deal with for a slightly cleaner package structure.

Reply:
I think we should add downgrade instructions in the Community Docs. For the time being, the instructions can be similar to [the gist](https://gist.github.com/hanefi/3575a53f8b059b5bdc8020d938a43bc9#file-instructions), and later we can improve if we decide to create seperate `citus-downgrade` packages.

What I had in my mind was to create backport releases to the first migration version of every major version. Usually all the patch releases share the same migration version so this is quite easy to accomplish. Since all backports share the same sql version, having a downgrade path to a single version is enough.

> when after upgrading from 9.2.0 to 9.3.0 you can actually downgrade to 9.2.0, instead of having to downgrade to 9.2.1, which contains the downgrade scripts from 9.3.0.

Nice point. However we should want the user to use the latest patch version, so I don't think we should concern ourselves with downgrades to older versions that were preceded by some bugfixes in a patch release.

> No need to do a lot of cherry picks to backport these changes to old releases. People can just install the shared ~citus-backport~ citus-downgrade package for the version they need to backport from.

I'd like the package name citus-downgrade more than citus-backport.

If we had a seperate package for downgrades, we will have a list of steps like the following:
1. Install `citus-downgrade` package for the current version
2. Run `ALTER EXTENSION citus UPDATE TO :older_version` (possibly precededd by a `SET citus.enable_version_checks TO 'false'`)
3. Install the citus package for :older_version (therefore uninstalling `citus-downgrade` as well due to package dependencies)
4. Restart PostgreSQL

This list of steps, in my opinion are slightly better than manually copying sql migration files manually.

Note: On the bad `9.2.3` release changes https://github.com/citusdata/citus/pull/3700
we removed `citus--9.2-2--9.3-1.sql`

Reply:
The steps you mention won't work if the user cannot start postgres because of something wrong with the new citus version. Having the following steps instead would solve this problem, as well as being easier and less error-prone to execute by the user:

1. Reinstall the `citus` package for :older_version (uninstalls `citus` package for new version)
2. Restart PostgreSQL
3. Run `ALTER EXTENSION citus UPDATE`

The way this can be done is make new `citus` package depend on the `citus-downgrade` package with the version being `>=` of the `citus` package. So you can have `citus` 9.2.0 with `citus-downgrade` 9.4.0, which contains the downgrade scripts from 9.4.0 to 9.3.0 and from 9.3.0 to 9.2.0.

That way the full flow will be:
1. Install `citus` 9.2.0, which in turn installs `citus-downgrade` 9.2.0
2. Run for couple of months
3. Install `citus` 9.4.0, which in turn installs `citus-downgrade` 9.4.0
4. Crashes happen
5. Install `citus` 9.2.0, which DOES NOT change the `citus-downgrade` version
6. Restart PostgreSQL
7. Run `ALTER EXTENSION citus UPDATE`.